### PR TITLE
Experimental: Use ASN1Decoder package

### PR DESCRIPTION
**Pull request by schwmi**
Instead of OpenSSL try to use a swift ASN1 parsing framework

# Open Issues:

- [x] Land ASN1 `signature` and `signatureAlgorithm` fix upstream
- [x] Use above properties accordingly:
  - [x] we currently use `signatureAlgorithm` instead of `signature`
  - [x] instead of `signatureAlgorithm` we use `rsaSignatureMessagePKCS1v15SHA1` hardcoded
- [x] Uncomment the lines in `verifyAuthenticity`
- [x] Clean-up no longer needed `CCrypto*` calls
- [x] check on https://github.com/filom/ASN1Decoder/pull/26
- [x] check on https://github.com/filom/ASN1Decoder/pull/28
- [x] Find out why sometimes `signature.signatureData` is `nil` (/cc @schwmi ). This currently happens for 2 test cases:
  - [x] `testMindNodeMacReceiptParsingWithFullValidation`
  - [x] `testMindNodeiOSSandBoxReceipt2ParsingAndValidation`

### CODE REVIEW

**Comment by schwmi** [comment]
Yes, swift for ASN1 parsing, and `swift-crypto` for has checking and receipt validation. This means that on apple platforms the crypto code falls back to `CryptoKit` and on Linux platforms it uses BoringSSL. 
Test are green, but the cert validation code is still commented out so no throw which leads to the test pass

**Comment by hannesoid** [comment]
Ok, so the open step is still certificate validation, for which we need two fixes for the now used ASN1 dependency where @schwmi @weichsel know what to do :)

**Comment by schwmi** [comment]
I am currently validating this receipt parsing logic on our API - comparing the results with the content from Apple's `/verifyReceipt` endpoint. If the results look good we should be able to merge this one.

**Comment by weichsel** [comment]
> Compared ~ 60k receipt parsings with the /verifyReceipt endpoint without any error. So looks good 👍
Any concerns regarding our usage of `SecKeyVerifySignature` - which is currently not tested via endpoint?

**Comment by schwmi** [comment]
Think for testing the signature verification process our unit tests should be enough.

**Comment by hannesoid** [comment]
I think this is in a great state now. I'm only wondering how to proceed with merging.
- some people (including us) might still use `experiment/openSSLXCFramework` branch currently, so maybe we should leave that dangling or merge, but keep the branch around, and then describe in the README what to do?
- we need a major version updated tag
- we need an updated README for how to integrate, especially describing how to use older release-tags for compatibility for people who can't immediately update to SPM for whatever reason

**Comment by hannesoid** [comment]
- [ ] check if we can move away from the ASN1Decoder `bugfix/intParsing` branch

**Comment by hannesoid** [comment]
Before Merging
- [ ] Update `README.md`:  
  - update integration/usage instructions
  - refer to older release tags for legacy version which uses Carthage not SPM
  - refer to `experiment/openSSLXCFramework` branch as legacy version for older Xcode versions without SPM
- [ ] Update `LICENSE`  
  - remove OpenSSL License
  
After Merging
- [ ] Add major version release tag

**Comment by schwmi** [comment]
@hannesoid Updated Readme and License, pls approve if we are ready for merge

File: LICENSE
Comment (hannesoid) [inline LICENSE:200]:
```suggestion
    See the License for the specific language governing permissions and
    limitations under the License.
```

**Review by hannesoid — APPROVED** [review]
I changed the base to `main`, imo ready to merge @schwmi

**Pull request merged**
